sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/ConversationAnalysisClient.cs

```diff
@@ -150,5 +153,136 @@ public virtual Response<AnalyzeConversationTaskResult> AnalyzeConversation(strin
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> to cancel the request.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="input"/> or <paramref name="tasks"/> or is null.</exception>
+        /// <exception cref="RequestFailedException">The service returned an error. The exception contains details of the service error.</exception>
+        public virtual async Task<Response<AnalyzeConversationJobState>> AnalyzeConversationAsync(IEnumerable<GeneratedConversation> input, IEnumerable<AnalyzeConversationLROTask> tasks, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(input, nameof(input));
+            Argument.AssertNotNull(tasks, nameof(tasks));
+
+            MultiLanguageConversationAnalysisInput multiLanguageConversationAnalysisInput = new MultiLanguageConversationAnalysisInput(input);
+
+            var analyzeConversationJobsInput = new AnalyzeConversationJobsInput(multiLanguageConversationAnalysisInput, tasks);
+
+            using DiagnosticScope scope = Diagnostics.CreateScope($"{nameof(ConversationAnalysisClient)}.{nameof(AnalyzeConversation)}");
+            scope.Start();
+
+            try
+            {
+                var responseHeaders = await _analysisRestClient.SubmitJobAsync(analyzeConversationJobsInput, cancellationToken).ConfigureAwait(false);
+                var jobId = getJobId(responseHeaders.Headers.OperationLocation);
+
+                await WaitForJobUntilDoneAsync(jobId).ConfigureAwait(false);
+
+                return await _analysisRestClient.JobStatusAsync(jobId, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        private async Task<bool> WaitForJobUntilDoneAsync(Guid jobId, int timeoutInSeconds = 0, int pollingIntervalInSeconds = 5)
+        {
+            var startTimeStamp = DateTime.Now;
+            while (true)
+            {
+                var jobStatus = await _analysisRestClient.JobStatusAsync(jobId).ConfigureAwait(false);
+
+                // check job status is done
+                if (jobStatus.Value.Status == JobState.Succeeded || jobStatus.Value.Status == JobState.Failed || jobStatus.Value.Status == JobState.PartiallySucceeded)
+                    return true;
+
+                // check for timeouts
+                if (timeoutInSeconds > 0)
+                {
+                    var nowTimeStamp = DateTime.Now;
+                    var diff = nowTimeStamp.Subtract(startTimeStamp).TotalSeconds;
+                    if (diff > timeoutInSeconds)
+                    {
+                        break;
+                    }
+                }
+
+                // sleep for a while
+                Thread.Sleep(pollingIntervalInSeconds);
+            }
+            return false;
+        }
```

Comment:
Not that this would be needed since we should return an LRO, but don't duplicate private code liek this. Instead, have a single `WaitForJobUntilDoneAsync` that takes a `bool isAsync` and awaits were necessary. You can see examples of this in our other SDKs like https://cs.github.com/Azure/azure-sdk-for-net/blob/bff99027890a8367c4fb99bf2347389953808784/sdk/keyvault/samples/keyvaultproxy/src/Cache.cs?q=isAsync+path%3Asdk%2Fkeyvault+-path%3Atests#L38. your async caller would call it + `ConfigureAwait(false)` while your sync caller would call it with `EnsureCompleted()`.